pallets/issue/src/tests.rs

```diff
@@ -624,3 +624,338 @@ fn test_request_issue_reset_interval_and_succeeds_with_rate_limit() {
+			VAULT3.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id3));
+
+		let issue_amount4 = 3000;
+		let issue_fee4 = 400;
+		let griefing_collateral4 = 1;
+		let amount_transferred4 = 3000;
+		let issue_id4 = setup_execute_with_vault(
+			issue_amount4,
+			issue_fee4,
+			griefing_collateral4,
+			amount_transferred4,
+			VAULT4.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id4));
+
+		let issue_amount5 = 3000;
+		let issue_fee5 = 700;
+		let griefing_collateral5 = 1;
+		let amount_transferred5 = 3000;
+		let issue_id5 = setup_execute_with_vault(
+			issue_amount5,
+			issue_fee5,
+			griefing_collateral5,
+			amount_transferred5,
+			VAULT5.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id5));
+	}
+
+	#[test]
+	fn integration_multiple_vaults_many_collateral() {
+		run_test(|| {
+			//set up the vaults
+			let collateral1: u128 = 1000;
+			register_vault_with_collateral(&VAULT, collateral1);
+
+			let collateral2: u128 = 2000;
+			register_vault_with_collateral(&VAULT2, collateral2);
+
+			let collateral3: u128 = 1500;
+			register_vault_with_collateral(&VAULT3, collateral3);
+
+			let collateral4: u128 = 1700;
+			register_vault_with_collateral(&VAULT4, collateral4);
+
+			let collateral5: u128 = 800;
+			register_vault_with_collateral(&VAULT5, collateral5);
+
+			//execute issues
+			let issue_asset = VAULT.wrapped_currency();
+			let issue_asset2 = VAULT2.wrapped_currency();
+			let issue_asset3 = VAULT3.wrapped_currency();
+			execute_multiple_issues();
+
+			//ensure that the current rewards equal to fee
+			//Example: we expect for reward(vault1, asset1) = floor(
+			//floor((1000/3000)*floor( 1400*(3000*5/(3000*5+3200*10+800*15)) )) = 118
+
+			//assertions for asset1 = asset4
+			assert_eq!(get_reward_for_vault(&VAULT, issue_asset), 118);
+			//NOTE: withouth floor(), should actually exactly be 237.25. In this case
+			//it is 236. The discrepancy is due to two consecutive roundings
+			assert_eq!(get_reward_for_vault(&VAULT2, issue_asset), 236);
+			assert_eq!(get_reward_for_vault(&VAULT3, issue_asset), 355);
+			assert_eq!(get_reward_for_vault(&VAULT4, issue_asset), 402);
+			assert_eq!(get_reward_for_vault(&VAULT5, issue_asset), 284);
+
+			//assertions for issue asset2 = asset5
+			assert_eq!(get_reward_for_vault(&VAULT, issue_asset2), 101);
+			assert_eq!(get_reward_for_vault(&VAULT2, issue_asset2), 202);
+			assert_eq!(get_reward_for_vault(&VAULT3, issue_asset2), 304);
+			assert_eq!(get_reward_for_vault(&VAULT4, issue_asset2), 345);
+			assert_eq!(get_reward_for_vault(&VAULT5, issue_asset2), 243);
+
+			//assertions for issue asset3
+			assert_eq!(get_reward_for_vault(&VAULT, issue_asset3), 50);
+			assert_eq!(get_reward_for_vault(&VAULT2, issue_asset3), 101);
+			assert_eq!(get_reward_for_vault(&VAULT3, issue_asset3), 152);
+			assert_eq!(get_reward_for_vault(&VAULT4, issue_asset3), 172);
+			assert_eq!(get_reward_for_vault(&VAULT5, issue_asset3), 122);
```

Comment:
The values for `VAULT3`, `VAULT4` and `VAULT5` don't make sense considering their collaterals defined in the preceding lines, do they? How can `VAULT3` always have more reward than `VAULT2` if it provides less collateral?

Reply:
That is because since the fee in the model are distributed also given the value of the pool in USD. In the mock for this tests defined [here](https://github.com/pendulum-chain/spacewalk/pull/428/files#diff-d6cdd749af964b12aacc03febe5032c67c92157c7f9864bdaec757186d686165R272), the value of collateral for 3,4 and 5 is higher in USD.